frontend/src/Page/Preparation.elm

```diff
@@ -599,6 +627,99 @@ uploadedIdentifierToLink identifier =
 
 
 
+-- ###################################################################
+-- CIP-100 Verification API
+-- ###################################################################
+
+
+{-| Convert IPFS URL to HTTPS gateway URL for the verification API.
+The CIP-100 verification API only accepts HTTPS URLs.
+-}
+ipfsToHttpsUrl : String -> String
+ipfsToHttpsUrl url =
+    if String.startsWith "ipfs://" url then
+        let
+            cid =
+                String.dropLeft 7 url
+        in
+        "https://ipfs.io/ipfs/" ++ cid
+
+    else
+        url
+
+
+{-| Call the CIP-100 verification API for a given metadata URL.
+Uses the backend proxy to avoid CORS issues.
+-}
+verifyCip100Metadata : String -> String -> Cmd Msg
+verifyCip100Metadata actionId metadataUrl =
+    let
+        -- Convert IPFS URLs to HTTPS gateway URLs
+        httpsUrl =
+            ipfsToHttpsUrl metadataUrl
+
+        verificationApiUrl =
+            "https://verifycardanomessage.cardanofoundation.org/api/verify-cip100?url=" ++ Url.percentEncode httpsUrl
+
+        proxyRequestBody =
+            JE.object
+                [ ( "url", JE.string verificationApiUrl )
```

Comment:
I’m curious, why not make the request directly? Is it because of CORS? Because any request we send through the server before going to final destination double the traffic and the server costs. If it’s just CORS, we should modify the verifycardanomessage server to accept direct request from the domains we use for the voting tool. Also, let’s move that into the Api module, which contains all requests of the website.